### PR TITLE
premake 5.0.0-beta2

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
We're already on `beta1`, though... (Not sure how that happened.) So `beta2` must be an improvement? At least, usually newer betas are more stable than older ones.

**Comment by chenrui333** [comment]
yeah, I think we probably like to follow the beta release (but I am happy to create another issue nudging them to make it official, it has been two years)

**Review by carlocab — APPROVED** [review]

**Comment by Bo98** [comment]
The bundled libzip is really old and needs https://github.com/nih-at/libzip/commit/70629b26c0ad0a709c444921574f49454850dea0.

File: Formula/premake.rb [hunk 1]
```diff
@@ -23,7 +23,14 @@ class Premake < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "f18473b6bf3eb65b2b8e36f35e2166bfe6f859abae5e63806f7bcb5ec675f2bf"
   end
 
+  on_linux do
+    depends_on "util-linux" # for uuid
+  end
+
   def install
+    # Workaround for Xcode 14.3
+    ENV.append_to_cflags "-Wno-implicit-function-declaration" if DevelopmentTools.clang_build_version >= 1403
```
Comment (carlocab) [inline Formula/premake.rb:32]: Can we use the patch at https://github.com/nih-at/libzip/commit/70629b26c0ad0a709c444921574f49454850dea0 instead?

Comment (chenrui333) [reply Formula/premake.rb:32]: only that patch is not enough

```
../../contrib/libzip/zip_close.c:621:2: error: call to undeclared function 'close'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        close(tfd);
        ^
zutil.c
```

**Comment by chenrui333** [comment]
There is some upstream discussion about the dependency updates stuff, https://github.com/premake/premake-core/issues/2092

File: Formula/premake.rb (hunk 1 above)
Comment (Bo98) [reply Formula/premake.rb:33]: Looks like they don't pass `-DHAVE_UNISTD_H` correctly, so we might need to do that ourselves as well until upstream fix it.

**Comment by chenrui333** [comment]
i would say that we can go for the merge and I can apply the further patches when upstream has incorporated the libzip side changes (I have already attach the issue reference in the code)

**Review by p-linnane — APPROVED** [review]

**Pull request merged**